### PR TITLE
Remove jasmine from test suite

### PR DESCRIPTION
**Pull request by nicolasstucki**

### CODE REVIEW

**Comment by nicolasstucki** [comment]
Review by @sjrd.

File: test-suite/js/src/test/scala/org/scalajs/testsuite/utils/PlatformTest.scala [hunk 1]
```diff
@@ -1,21 +0,0 @@
...
-
-import org.scalajs.jasminetest.JasmineTest
-
-object PlatformTest extends JasmineTest {
-  when("typedarray").
-  describe("org.scalajs.testsuite.utils.Platform") {
-    it("areTypedArraysSupported should be true") {
-      // Note that if we don't have the tag set we can't say anything
-      // on the typed arrays support.
-      expect(Platform.areTypedArraysSupported).toBeTruthy
```
Comment (sjrd) [inline test-suite/js/src/test/scala/org/scalajs/testsuite/utils/PlatformTest.scala]: Is there any other test that performs this check?

File: javalib-ex-test-suite/src/test/scala/scala/scalajs/testsuite/javalib/lang/ObjectTestEx.scala [hunk 2]
```diff
@@ -7,42 +7,40 @@
...
-      expect(o2.getClass.asInstanceOf[js.Any]).toBe(classOf[SomeCloneable].asInstanceOf[js.Any])
-      expect(o2.x).toEqual(5)
-      expect(o2.y()).toEqual(8)
+      @noinline def y(): Int = x + 3
     }
 
+    val o = new SomeCloneable(5)
+    val o2 = o.clone()
+    assertNotSame(o.asInstanceOf[js.Any], o2.asInstanceOf[js.Any])
+    assertSame(classOf[SomeCloneable].asInstanceOf[js.Any], o2.getClass.asInstanceOf[js.Any])
```
Comment (sjrd) [inline javalib-ex-test-suite/src/test/scala/scala/scalajs/testsuite/javalib/lang/ObjectTestEx.scala]: The casts in these two lines are not needed anymore.

File: javalib-ex-test-suite/src/test/scala/scala/scalajs/testsuite/javalibex/DataInputStreamTest.scala [hunk 3]
```diff
@@ -2,329 +2,352 @@ package scala.scalajs.testsuite.javalibex
...
+        0x00, 0x00, 0x00, 0x00,
+        0x7f, 0xff, 0xff, 0xff,
+        0xff, 0xff, 0xff, 0xfc,
+        0x00, 0x00, 0x00, 0x53,
+        0x00, 0x00, 0x89, 0xa2,
+        0xff, 0xfe, 0x82, 0xfd,
+        0x80, 0x00, 0x00, 0x00,
+        0x00, 0x00, 0x00, 0x01)
+
+    assertEquals(0, stream.readInt(), 0)
```
Comment (sjrd) [inline javalib-ex-test-suite/src/test/scala/scala/scalajs/testsuite/javalibex/DataInputStreamTest.scala]: What is the third argument when comparing `Int`s?
(same below with `Long`, `Short`, etc.)

File: javalib-ex-test-suite/src/test/scala/scala/scalajs/testsuite/utils/AssertThrows.scala [hunk 4]
```diff
@@ -0,0 +1,68 @@
+package scala.scalajs.testsuite.utils
+
+/** This is a copy of the implementation in the testSuite */
+object AssertThrows {
+  /** Backport implementation of Assert.assertThrows to be used until JUnit 4.13 is
+    *  released. See org.junit.Assert.scala in jUnitRuntime.
+    */
```
Comment (sjrd) [inline javalib-ex-test-suite/src/test/scala/scala/scalajs/testsuite/utils/AssertThrows.scala]: Scaladoc style.

File: javalib-ex-test-suite/src/test/scala/scala/scalajs/testsuite/utils/AssertThrows.scala [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+  private def assertThrowsBackport(expectedThrowable: Class[_ <: Throwable],
+                                   runnable: ThrowingRunnable): Unit = {
```
Comment (sjrd) [inline javalib-ex-test-suite/src/test/scala/scala/scalajs/testsuite/utils/AssertThrows.scala]: 4 spaces

File: javalib-ex-test-suite/src/test/scala/scala/scalajs/testsuite/utils/AssertThrows.scala [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+      try {
+        runnable.run()
+        null.asInstanceOf[T]
+      } catch {
+        case actualThrown: Throwable =>
+          if (expectedThrowable.isInstance(actualThrown)) {
+            actualThrown.asInstanceOf[T]
+          } else {
+            val mismatchMessage = "unexpected exception type thrown;" +
+                expectedThrowable.getSimpleName + " " + actualThrown.getClass.getSimpleName
```
Comment (sjrd) [inline javalib-ex-test-suite/src/test/scala/scala/scalajs/testsuite/utils/AssertThrows.scala]:
```scala
val mismatchMessage = {
  "unexpected exception type thrown;" +
  expectedThrowable.getSimpleName + " " + actualThrown.getClass.getSimpleName
}
```

**Comment by sjrd** [comment]
That's all.

**Comment by sjrd** [comment]
Dear Jenkins, retest this please

File: test-suite/js/src/test/scala/org/scalajs/testsuite/utils/PlatformTest.scala (hunk 1 above)
Comment (nicolasstucki) [inline test-suite/js/src/test/scala/org/scalajs/testsuite/utils/PlatformTest.scala]: I thought so, but the tests that I had in mind do not cover this case. But now it is possible to write this in JUnit, so i will do that.

Comment (nicolasstucki) [inline test-suite/js/src/test/scala/org/scalajs/testsuite/utils/PlatformTest.scala]: I also found a simple way to keep the `JUnitBootstrapTest`.

File: javalib-ex-test-suite/src/test/scala/scala/scalajs/testsuite/utils/AssertThrows.scala [hunk 7]
```diff
@@ -0,0 +1,70 @@
+package scala.scalajs.testsuite.utils
+
+/** This is a copy of the implementation in the testSuite */
+object AssertThrows {
+  /** Backport implementation of Assert.assertThrows to be used until JUnit 4.13 is
+   *  released. See org.junit.Assert.scala in jUnitRuntime.
+   */
+  private def assertThrowsBackport(expectedThrowable: Class[_ <: Throwable],
+                                   runnable: ThrowingRunnable): Unit = {
```
Comment (sjrd) [inline javalib-ex-test-suite/src/test/scala/scala/scalajs/testsuite/utils/AssertThrows.scala]: 4 spaces here too

**Comment by sjrd** [comment]
That's all.

**Comment by sjrd** [comment]
LGTM

File: ci/matrix.xml [hunk 8]
```diff
@@ -61,6 +61,7 @@
         'set scalaJSOptimizerOptions in testingExample ~= (_.withDisableOptimizer(true))' \
         ++$scala testingExample/test:run testingExample/test &&
     sbtretry ++$scala testSuiteJVM/test testSuiteJVM/clean &&
+    sbtretry ++$scala testSuite/test:runMain org.scalajs.testsuite.junit.JUnitBootstrapTest testSuite/clean &&
```
Comment (sjrd) [inline ci/matrix.xml]: As the CI failure highlights, there should be `'`s around `testSuite/test:runMain org.scalajs.testsuite.junit.JUnitBootstrapTest`. So this line should be:
```
    sbtretry ++$scala 'testSuite/test:runMain org.scalajs.testsuite.junit.JUnitBootstrapTest' testSuite/clean &&
```

**Comment by sjrd** [comment]
LGTM

**Pull request merged**